.github/workflows/cmake.yml

```diff
@@ -0,0 +1,360 @@
+name: Build
+
+on: [push]
+
+jobs:
+  linux-gcc:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        VER: [7, 8, 9]
+        EXT: [ON, OFF]
+        GEN: [Unix Makefiles, Ninja]
+        VAR: [Debug, Release]
+        STD: [11] # 90 results in errors
+        BIN: [32, 64]
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Create Build Environment
+      run: cmake -E make_directory $GITHUB_WORKSPACE/build;
+        cmake -E make_directory $GITHUB_WORKSPACE/install;
+        cmake -E make_directory $GITHUB_WORKSPACE/downstream/pkgconfig/bare;
+        cmake -E make_directory $GITHUB_WORKSPACE/downstream/pkgconfig/sdk;
+        if [[ "${{matrix.GEN}}" == "Ninja" && ! `which ninja` ]]; then sudo apt install -y ninja-build; fi;
+        if [[ "${{matrix.BIN}}" == "32" ]]; then sudo apt install -y gcc-multilib gcc-8-multilib gcc-9-multilib; fi
+      # Install Ninja only if it's the selected generator and it's not available.
+
+    - name: Configure CMake
+      shell: bash
+      run: cmake
+        -G "${{matrix.GEN}}"
+        -D BUILD_TESTING=ON
+        -D CMAKE_BUILD_TYPE=${{matrix.VAR}}
+        -D CMAKE_C_FLAGS="-Wall -Wextra -Werror -pedantic -Wno-format -m${{matrix.BIN}}"
```

Comment:
We'll probably want to fix format warnings eventually. Would be good to have an issue for this.

Reply:
#132 written out.

Reply:
Thanks!